app/actions/views/channel.js

```diff
@@ -195,10 +196,21 @@ export function unmarkFavorite(channelId) {
 export function leaveChannel(channel, reset = false) {
     return async (dispatch, getState) => {
         const {currentId: teamId} = getState().entities.teams;
-        serviceLeaveChannel(teamId, channel.id)(dispatch, getState).then(() => {
-            if (channel.isCurrent || reset) {
-                selectInitialChannel(teamId)(dispatch, getState);
-            }
-        });
+        await serviceLeaveChannel(teamId, channel.id)(dispatch, getState);
+        if (channel.isCurrent || reset) {
+            await selectInitialChannel(teamId)(dispatch, getState);
+        }
+        if (reset) {
+            dispatch({type: NavigationTypes.NAVIGATION_POP}, getState);
+        }
+    };
+}
+
+export function deleteChannel(teamId, channelId, shouldGoBack = false) {
```

Comment:
Since we're only using this on one place, I don't think we need a new action here. You can just have
```
this.props.actions.deleteChannel(teamId, channelId).then(this.props.actions.goBack)
```
instead adding this new function with a flag that is currently always true.